src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -232,48 +225,8 @@ protected void DecorateItemsWithMetadata(ICollection<I> items, ImmutableList<Pro
             /// </summary>
             protected ICollection<I> SelectItemsMatchingItemSpec(ImmutableList<ItemData>.Builder listBuilder, IElementLocation elementLocation)
```

Comment:
Not directly related to this PR, but why does this take a list builder instead of just a list?